src/parser/shaman/elemental/modules/talents/Ascendance.js

```diff
@@ -35,36 +52,94 @@ class Ascendance extends Analyzer {
     return this.rawUpdate;
   }
 
+  get averageLavaBurstCasts() {
+    return this.numCasts[SPELLS.LAVA_BURST.id]/this.numCasts[SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id];
+  }
+
   on_byPlayer_cast(event) {
-    if (this.selectedCombatant.hasBuff(SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id, event.timestamp)) {
-      const spellId = event.ability.guid;
-      if (this.numCasts[spellId] !== undefined) {
-        this.numCasts[spellId] += 1;
-      } else {
-        this.numCasts.others += 1;
+    const spellId = event.ability.guid;
+    const target = this.enemies.getEntity(event);
+
+
+    if(spellId === SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id) {
+      this.justEnteredAcendance = true;
+      this.numCasts[SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id]++;
+    }
+
+    if(this.justEnteredAcendance){
```

Comment:
This logic is somewhat weird. Is the point of this that if the Flame Shock would need to be refreshed during Ascendance (its duration is less than the Ascendance duration), then it's bad cast? If so, you might want to keep track of the Flame Shock duration manually, because as far as I know, `getBuffUptime()` aggregates all Flame Shocks on the target (basically sums up all durations before it as well). Please clarify a little bit.

Reply:
Wait so it sums up all flame shocks from different players?

Reply:
Should, unless you give the function a sourceID to work with `getBuffUptime(spellId, sourceID)`

Reply:
Even worse, I think it uses past Flame Shocks as well:
https://github.com/WoWAnalyzer/WoWAnalyzer/blob/e73129905e5bb4832849d439decb6ed796dd4624/src/parser/core/Entity.js#L81
But from the look of it, try `getBuff`, see what it returns - this line suggests it might have `start` property
https://github.com/WoWAnalyzer/WoWAnalyzer/blob/e73129905e5bb4832849d439decb6ed796dd4624/src/parser/core/Entity.js#L20